File: recursion and backtracking/convert.cpp
#include <iostream>
#include <string>
using namespace std;

string findBinary(int decimal, string result) {
    if(decimal==0) {
        return result;
    }

    result = to_string(decimal % 2) + result;
    return findBinary(decimal/2, result);
}

int main() {    
    string result = findBinary(233,"");
    cout << result;
    return 0;
}

File: recursion and backtracking/convert.java
public class DecimalToBinary {
    public static void main (string[] args) {
        string binary = findBinary(233, "");
    }
    public static string findBinary(int decimal, string result) {
        if (decimal == 0) {
            return result;
        }

        result = decimal % 2 + result;
        return findBinary(decimal/2, result);
    }
}
